packages/web-components/src/components/toggle/toggle.ts

```diff
@@ -111,7 +117,47 @@ class CDSToggle extends HostListenerMixin(CDSCheckbox) {
    * The text for the unchecked state.
    */
   @property({ attribute: 'label-b' })
-  labelB = '';
+  labelB = 'Off';
+
+  /**
+   * Private references of external <label> elements that are
+   * `for="this-toggle-element-id"`
+   */
+  private _externalLabels: HTMLLabelElement[] = [];
+
+  /**
+   * Handles `click` on external `<label>`
+   */
+  private _onExternalLabelClick = () => {
+    this._checkboxNode?.focus();
+    this._handleChange();
+  };
+
+  /**
+   * Finds external toggle `<label>`s and attaches handlers.
+   */
+  private _attachExternalLabels() {
+    const doc = this.ownerDocument || document;
+    const found: HTMLLabelElement[] = [];
+
+    if (this.id) {
+      found.push(
+        ...(Array.from(
+          doc.querySelectorAll(`label[for="${this.id}"]`)
+        ) as HTMLLabelElement[])
+      );
+    }
```

Comment:
```suggestion
    const found = this.id
      ? [...doc.querySelectorAll<HTMLLabelElement>(`label[for="${this.id}"]`)]
      : [];
    }
```

`querySelectorAll` already returns a list, so you probably don't need the `Array.from`. And using `<HTMLLabelElement>` lets you avoid the manual casting.